fe1-web/network/NetworkConnection.ts

```diff
@@ -61,10 +53,28 @@ export class NetworkConnection {
           throw new NetworkError(`Received a response which id = ${response.id} does not match any pending requests`);
         }
 
+        // a response was received, we clear the timeout and the pending query from the pending map
+        clearTimeout(pendingResponse.timeoutId);
         this.payloadPending.delete(response.id);
 
-        pendingResponse.resolvePromise(response);
-        this.onMessageHandler(response);
+        // reject the promise if the response is negative
+        if (response.isPositiveResponse()) {
+          pendingResponse.resolvePromise(response);
+          try {
+            this.onMessageHandler(response);
```

Comment:
do we need a handler at all here if the promise gets resolved?

Reply:
So. In the last version (see last commit), I left it in. In my opinion, some JsonRpcResponse could need a handler (e.g. store something in localStorage)